lib/ex_unit/test/ex_unit/capture_io_test.exs

```diff
@@ -86,6 +86,102 @@ defmodule ExUnit.CaptureIOTest do
            end) == "a"
   end
 
+  test "async capture_io works with put chars to stderr" do
+    me = self()
+
+    Enum.each(1..6, fn num ->
+      spawn(fn ->
+        captured =
+          capture_io(:stderr, fn ->
+            :io.put_chars(:standard_error, "#{num}\n")
+            Process.sleep(10)
+            :io.put_chars(:standard_error, "#{num + 10}\n")
+          end)
+
+        send(me, captured)
+      end)
+
+      Process.sleep(1)
+    end)
+
+    expected = [
+      {~r/^1\n/, ~r/11\n/},
+      {~r/^2\n/, ~r/12\n/},
+      {~r/^3\n/, ~r/13\n/},
+      {~r/^4\n/, ~r/14\n/},
+      {~r/^5\n/, ~r/15\n/},
+      {~r/^6\n/, ~r/16\n/}
+    ]
```

Comment:
Let's avoid regexes if possible. We could for example log them as "before:#{num}" and "after:#{num}" and use `String.contains?` (or continue using `=~` which also works for binaries).